pyproject.toml

```diff
@@ -123,51 +121,65 @@ exclude = [
+    "D103",     # pydocstyle: undocumented-public-function
+    "D417",     # pydocstyle: undocumented-param
+    "B023",     # flake8-bugbear: function-uses-loop-variable
+]
+ignore = [
+    "PLR0913",  # pylint: too-many-arguments                            # TODO: activate this
+    "PLR2004",  # pylint: magic-value-comparison                        # TODO: activate this
+    "PT009",    # flake8-pytest-style: pytest-unittest-assertion        # TODO: activate when pytest is everywhere
+    "PT027",    # flake8-pytest-style: pytest-unittest-raises-assertion # TODO: activate when pytest is everywhere
+    "ISC001",   # flake8-implicit-str-concat: single-line-implicit-string-concatenation  # incompatible with the formatter
+]
 
 
-[tool.isort]
-profile = "black"
-line_length = 120
-
-
-[tool.pylint.MASTER]
-ignore-paths = ["src/dsp_tools/models/resource.py"] # TODO: activate this
-suggestion-mode = true
-
-
-[tool.pylint.format]
-max-line-length = 120
-
-
-[tool.pylint."messages control"]
-disable = [
-    "raw-checker-failed",            # disabled by default
-    "bad-inline-option",             # disabled by default
-    "locally-disabled",              # disabled by default
-    "file-ignored",                  # disabled by default
-    "suppressed-message",            # disabled by default
-    "deprecated-pragma",             # disabled by default
-    "use-symbolic-message-instead",  # disabled by default
-    "c-extension-no-member",         # probably not necessary
-    "use-dict-literal",              # not worth the effort
-    "use-list-literal",              # not worth the effort
-    "missing-module-docstring",      # not worth the effort
-    "trailing-whitespace",           # not worth the effort
-    "use-maxsplit-arg",              # not worth the effort
-    "global-statement",              # not worth the effort
-    "no-else-continue",              # would be less readable
-    "no-else-return",                # would be less readable
-    "no-else-raise",                 # would be less readable
-    "logging-fstring-interpolation", # would be less readable
-    "duplicate-code",                # too many false positives
-    "invalid-name",                  # TODO: activate this
-    "too-many-arguments",            # TODO: activate this
-    "too-many-branches",             # TODO: activate this
-    "too-many-lines",                # TODO: activate this
-    "too-many-locals",               # TODO: activate this
-    "too-many-nested-blocks",        # TODO: activate this
-    "too-many-return-statements",    # TODO: activate this
-    "too-many-statements",           # TODO: activate this
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = [
+    "F401",     # unused-import
+    "F403",     # undefined-local-with-import-star
+    "PLC0414",  # useless-import-alias
+]
+"src/dsp_tools/import_scripts/import_script.py" = [
+    "PLW2901",  # pylint: redefined-loop-name 
```

Comment:
It is correct that here we need a pylint ignore?

Reply:
Ruff reimplements rules from many linters, so the historical origin of a rule is often mentioned. Here, "PLW2901" means that this rule belongs to the group of reimplemented Pylint Warnings (hence PLW). 

Concerning my comment `# pylint: redefined-loop-name`: I need a human-friendly name, so I decided to make comments of the pattern `# original-linter: rule-name`.

Reply:
So ruff understands pylint syntax? or this is solely for us? If it is solely for us, I may stay away from specifying with pylint. Because if they change, then we may be tempted to go to pylint and not read the ruff docs. Maybe you can comment it in your own language?

Reply:
Yes, it is solely for us. The reason I commented every rule with the name of its ruleset is the section `[tool.ruff.lint] > select`: Here, I need a way to describe each ruleset, so I wrote 

```
"PL",       # pylint`
```

But arguably, when ignoring single rules, the ruleset name could be omitted. So instead of 

```
ignore = [
    "PLR0913",  # pylint: too-many-arguments
    "PLR2004",  # pylint: magic-value-comparison
    "PT009",    # flake8-pytest-style: pytest-unittest-assertion 
    "PT027",    # flake8-pytest-style: pytest-unittest-raises-assertion
    "ISC001",   # flake8-implicit-str-concat: single-line-implicit-string-concatenation
]
```

I could also do this: 

```
ignore = [
    "PLR0913",  # too-many-arguments
    "PLR2004",  # magic-value-comparison
    "PT009",    # pytest-unittest-assertion 
    "PT027",    # pytest-unittest-raises-assertion
    "ISC001",   # single-line-implicit-string-concatenation
]
```

Would you prefer the second over the first?

Reply:
If ruff is implementing flake8 then I would be fine with the flake annotation.
If they are implementing their own rule, which is alike another, then I prefer the second.
Thanks!